lib/task/import/csvCheckImportTask.class.php

```diff
@@ -21,172 +21,152 @@
  * Check csv data.
  *
  * @author     Mike Cantelon <mike@artefactual.com>
+ * @author     Steve Breker <sbreker@artefactual.com>
  */
-class csvCheckImportTask extends csvImportBaseTask
+class csvCheckImportTask extends arBaseTask
 {
-    protected $namespace = 'csv';
-    protected $name = 'check-import';
-    protected $briefDescription = 'Check CSV data, providing diagnostic info';
-
-    protected $detailedDescription = <<<'EOF'
-Check CSV data, providing information about it
+  protected $verbose = null;
+
+  protected function configure()
+  {
+    $this->addArguments(array(
+      new sfCommandArgument('filename', sfCommandArgument::REQUIRED,
+        'The input file name (csv format).')
+    ));
+
+    $this->addOptions(array(
+      new sfCommandOption('application', null,
+        sfCommandOption::PARAMETER_OPTIONAL, 'The application name', 'qubit'
+      ),
+      new sfCommandOption('env', null, sfCommandOption::PARAMETER_REQUIRED,
+        'The environment', 'cli'
+      ),
+      new sfCommandOption('connection', null,
+        sfCommandOption::PARAMETER_REQUIRED, 'The connection name', 'propel'
+      ),
+      new sfCommandOption('verbose', 'i',
+        sfCommandOption::PARAMETER_NONE,
+        'Provide detailed information regarding each test'
+      ),
+    ));
+
+    $this->namespace = 'csv';
+    $this->name = 'check-import';
+    $this->briefDescription = 'Check CSV data, providing diagnostic info.';
+    $this->detailedDescription = <<<EOF
+    Check CSV data, providing information about it.
 EOF;
+  }
+
+  /**
+   * @see sfTask
+   */
+  public function execute($arguments = array(), $options = array())
+  {
+    parent::execute($arguments, $options);
+    
+    $validatorOptions = $this->setOptions($options);
+
+    if (isset($options['verbose']) && $options['verbose'])
```

Comment:
Nitpick: could check for `!empty($options['verbose'])`.

Reply:
Nice - fixed.